starport/pkg/cosmosanalysis/app/app.go

```diff
@@ -0,0 +1,52 @@
+package app
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+)
+
+const (
+	appTypeName = "App"
```

Comment:
I think we should discover the app struct as well. It seems like every app needs to implement [servertypes.Application](https://github.com/cosmos/cosmos-sdk/blob/e306e5b/server/types/app.go#L34), we can discover the app struct by checking against this interface.

Reply:
Yes, it can be possible to discover what is the "App" in the file by checking the interface implementation like for Msg, I was actually thinking about it. The thing is, with a lazy check like this one we make the condition to register the module in app.go very restrictive (in the sense we prevent registering the module if the user has customized app.go too much) App struct name could be changed but if this struct has been changed it can be potentially impossible to save the module dependency with our current method by providing app.FooKeeper instance to the keeper of the module to register (app variable name could be changed as well)
If the check fails, the module is still successfully generated, it will just not be registered in app.go. It's better to have a failed check than a less restrictive check but that can potentially generate an app that will not compile

Reply:
Sorry, I couldn't get what you mean. I'm just suggesting to find the name of app struct with code analysis, you mean there can be a downside of this? I'm no sure how assuming a constant `App` name makes things more restrictive, I don't think it's a valid indicator of checking if user changed too much in a starport scaffold chain.